### DISCUSSION TITLE
Add `toQueryParams()` and `fromQueryParams()` helpers

### DISCUSSION THREAD

**Discussion by NixBiks** [TanStack/table#3967] [Ideas]
I wonder how many are actually looking at saving state in url query parameters. I wonder if we should add some helpers for this? I'm not exactly sure how to due it - but I guess we want to translate serialize and deserialize table state. 

However it is not enough since we'd also need "serialize next state if I click here" sorta thing. If anyone have some good ideas on how to solve this then please jot down your ideas. I can do a PR once I've figured out how to solve it.

**Comment by tannerlinsley** [comment]
I agree this is a use case, but serialization is not within the scope of Table. I would instead look into your routers methods of serialization or check out React Location.

**Reply by NixBiks** [reply]
Fair game! I've actually never used react-location since I'm mostly using NextJS. As far as I know then there is not really a need for it then. 

But yeah I guess I'll come up with some method using the router from nextjs for now.
